### PR TITLE
Reporting Control Changes

### PR DESCRIPTION
**Pull request by paulxtiseo**
Added a bit more control over reporting module behavior for a project I am using retry-as-promised on, also added cross-env package (for testing on Windows and environment var setting) and updated Bluebird.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,13 +16,17 @@ module.exports = function retryAsPromised(callback, options) {
...
+    reportFn: options.reportFn || console.log,
+    name:  callback.name || options.name || 'unknown'
   };
   
   // Massage match option into array so we can blindly treat it as such later
   if (!Array.isArray(options.match)) options.match = [options.match];
 
-  debug('Trying '+callback.name+' (%s)', options.$current);
+  debug('Trying '+ options.name+' (%s)', options.$current);
+  if(options.report) options.reportFn('Trying ' + options.name + ' #' + options.$current + ' at ' + new Date().toLocaleTimeString());
```
Comment (mickhansen) [inline index.js]: To maintain bc lets scrap `reportFn` and just use `options.report` if defined.
And i think `options.report` should take two arguments, a log string and then meta data `{name, current, timestamp}`

**Comment by mickhansen** [comment]
README change for report would be great.

**Comment by paulxtiseo** [comment]
Mae changes per my interpretation of your request. Not sure it's what you meant? I changed it so that the function passed into RAP will be called and passed two params: a display string and the options object created by RAP. The developer can then have his function use that data however they wish.
Made the appropriate README changes too.

PS: Not sure if this requires new tetss?

**Comment by mickhansen** [comment]
v2.1.0

**Pull request merged**